starport/pkg/goanalysis/goanalysis_test.go

```diff
@@ -0,0 +1,19 @@
+package goanalysis
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFindTypes(t *testing.T) {
+	srcDirPath, _ := filepath.Abs(filepath.Join("..", "..", "services", "chain"))
```

Comment:
I don't think we should rely on parent directories for tests. I don't even think we should rely on the source code.

The way I would do is to have a custom source that I write in a temporary directory for parsing. We could have like two subdirectories with some files with types and interfaces for testing that are then used for all the methods that will be in this package.

Not sure this is a very interesting task if you want to learn, we can handle this if you want.